packages/rich-text/src/component/use-input-and-selection.js

```diff
@@ -132,6 +132,16 @@ export function useInputAndSelection( props ) {
 				onSelectionChange,
 			} = propsRef.current;
 
+			const isDisabled =
+				! element.contentEditable ||
+				element.contentEditable === 'false';
```

Comment:
Previously clicking a RichText was triggering a selection change even if the contentEditable was disabled.

Reply:
Yep, that was the same change with https://github.com/WordPress/gutenberg/pull/40611 😄